src/API/Google/MerchantReport.php

```diff
@@ -52,6 +56,89 @@ public function __construct( ShoppingContent $service, ProductHelper $product_he
 		$this->product_helper = $product_helper;
 	}
 
+	/**
+	 * Get ProductView Query response.
+	 *
+	 * @param string|null $next_page_token The next page token.
+	 * @return array Array of products along with their statuses.
+	 *
+	 * @throws Exception If the product view report data can't be retrieved.
+	 */
+	public function get_product_view_report( $next_page_token = null ): array {
+		$batch_size = apply_filters( 'woocommerce_gla_product_view_report_page_size', 1000 );
+
+		try {
+			$product_view_data = [
+				'statuses'  => [],
+				'next_page' => null,
+			];
+
+			$query = new MerchantProductViewReportQuery(
+				[
+					'next_page' => $next_page_token,
+					'per_page'  => $batch_size,
+				]
+			);
+
+			$response = $query
+			->set_client( $this->service, $this->options->get_merchant_id() )
+			->get_results();
+
+			$rows = $response->getResults();
+
+			foreach ( $rows as $row ) {
+
+				/** @var ProductView $product_view  */
+				$product_view = $row->getProductView();
+
+				$wc_product_id     = $this->product_helper->get_wc_product_id( $product_view->getId() );
+				$mc_product_status = $this->convert_aggregated_status_to_mc_status( $product_view->getAggregatedDestinationStatus() );
+
+				// Skip if the product id does not exist
+				if ( ! $wc_product_id ) {
+					continue;
+				}
+
+				$product_view_data['statuses'][] = [
+					'product_id'      => $wc_product_id,
+					'status'          => $mc_product_status,
+					'expiration_date' => $this->convert_shopping_content_date( $product_view->getExpirationDate() ),
+				];
+
+			}
+
+			$product_view_data['next_page'] = $response->getNextPageToken();
+
+			return $product_view_data;
+		} catch ( GoogleException $e ) {
+			do_action( 'woocommerce_gla_mc_client_exception', $e, __METHOD__ );
+			throw new Exception( __( 'Unable to retrieve Product View Report.', 'google-listings-and-ads' ), $e->getCode() );
```

Comment:
Just wondering, since this is going to be recorded as a failure in a Scheduled Action scheduler job, do you think it will be useful to include both `$e->getMessage` and `$e->getCode`? Otherwise they have to look up the matching timestamp to see more details in the log.
```suggestion
			throw new Exception( __( 'Unable to retrieve Product View Report: ', 'google-listings-and-ads' ) . $e->getMessage(), $e->getCode() );
```

Or format the error with sprintf if we want to allow for greater translation/RTL support.